docs/vNext/DeclarativeWorkloadModule.md

```diff
@@ -0,0 +1,190 @@
+    - [Formatted String](#formatted-string)
+
+## Overview
+`DeclarativeWorkloadModuleBase` is a Caliper `WorkloadModule` implementation that is used through assiging `declarative` to `workload.module` in the Benchmark Configuration file. The Contracts, Functions and Parameters for test runs are specified under `arguments.behavior`.
+
+## Example
+
+```yaml
+workload:
+    module: declarative
+        arguments:
+            parameterOne: param1
+            parameterTwo: 42
+            behavior:
+                contractSelection: 'uniform_random_list_element'
+                contracts:
+                - name: contract1
+                  functionSelection: 'uniform_random_list_element'
+                  functions:
+                  - name: function1
+                    parameters:
+                    - name: randomNumber
+                      type: uniform_random
+                      options:
+                        min: 10
+                        max: 100
+```
+The example above means the follows:
+- The WorkloadModule used here is `declarative`.
+- The `roundArguments` taken in by Caliper are `parameterOne`, assigned the value of `'param1'` and `parameterTwo` assigned the value of `42`.
+- The `arguments.behavior` section specifies the declared properties of the workload module.
+- `contractSelection` specifies how a contract will be chosen from a `contracts` list.
+- `contracts` contains `name` and `functionSelection`, followed by `functions` which has `function1` present in it as the only list item.
+- `functions` contains `name` and `parameters`.
+- `parameters` contains a value provider with the `name` randomNumber of `type` uniform_random. This generates a random number between 10 and 100 for the parameter. 
+
+## Hierarchy
+Under `arguments.behavior`, `contracts` is the top layer. It consists of a list of contracts to be tested. Within a single `contracts` list element, `functions` property holds the list of all functions present under that contract. Similarly, under each `functions` list item, there is a `parameters` list which has different types of user defined parameters under it.
+
+### Contracts
+
+Used to specify the list of contracts to be tested. Each `contracts` list element has the following format.
+
+|Property|Type       |Description|
+|:-------|:----------|:----------|
+|name    |string     |Name of SUT contract to be tested|
+|functionSelection|string|Type of contract picking logic|
+|function|list|List of Contract functions|
+
+### Functions
+
+Used to specify the list of functions under a contract to be tested. Each `functions` list element has the following format.
+
+|Property|Type|Description|
+|:-------|:---|:----------|
+|name    |string|Name of SUT function to be tested|
+|parameters|list|List of user-defined parameters to be generated|
+
+### Parameters
+
+Used to specify different generated parameters for each function. The `parameters` list can contain one or more of the following items.
```

Comment:
A table would be useful to describe the three parameters of a Parameter object: name, type, options. The two latter should delegate the explanation to the different value providers.